gui/src/app/pages/HomePage/LoadProjectWindow.tsx

```diff
@@ -49,25 +43,19 @@ const LoadProjectWindow: FunctionComponent<LoadProjectWindowProps> = ({
             files: fileManifest,
             clearExisting: replaceProject,
           });
-        } else if (
-          filesUploaded.length === 1 &&
-          filesUploaded[0].name.endsWith(".stan")
-        ) {
-          // a single .stan file
-          if (replaceProject) {
-            update({ type: "retitle", title: filesUploaded[0].name });
-          }
-          const fileManifest: Partial<FieldsContentsMap> = {
-            stanFileContent: parseFile(filesUploaded[0].content),
-          };
-          update({
-            type: "loadFiles",
-            files: fileManifest,
-            clearExisting: replaceProject,
-          });
         } else {
+          let stanFileName = "";
           const files: Partial<FileRegistry> = {};
+
           for (const file of filesUploaded) {
+            if (file.name.endsWith(".stan")) {
+              if (stanFileName !== "") {
+                throw Error("Only one .stan file can be uploaded at a time");
+              }
+              files["main.stan"] = parseFile(file.content);
+              stanFileName = file.name;
+              continue;
+            }
             if (!Object.values(FileNames).includes(file.name as any)) {
               throw Error(`Unrecognized file: ${file.name}`);
```

Comment:
Suppose the user uploaded multiple files including something.zip. This message would then say "Unrecognized file: something.zip" whereas it should really say: "When importing from a Zip file, you must only upload the one .zip file" or something to that effect.

Also, I'm not sure "Unrecognized file" is the right message here. Maybe, "Unsupported file name"

Reply:
Updated to make both changes